lines.js

```diff
@@ -35,34 +51,60 @@ function buttonState(e) {
         button1.innerHTML = "Turn on";
         document.getElementById("panels").classList.remove("released");
     }
+
 }
 
-function drawPoint(e) {
+function startDraw(e) {
     if (pressed) {
-        context.beginPath();
-        context.arc(e.pageX - myCanvas.offsetLeft, e.pageY - myCanvas.offsetTop, radius, 0, 2 * Math.PI);
-        context.closePath();
-        context.stroke();
-        context.fill();
+        if (currentIndex !== historyChanges) {
+            projectHistory.splice(currentIndex + 1);
+            historyChanges = currentIndex;
+        }
+        drawPoint(radius, e.pageX - myCanvas.offsetLeft, e.pageY - myCanvas.offsetTop);
         context.beginPath();
         context.moveTo(e.pageX - myCanvas.offsetLeft, e.pageY - myCanvas.offsetTop);
-        context.lineWidth = 2 * radius;
         mousePressed = true;
+        isOut = false;
     }
 }
 
 function drawLine(e) {
     if (pressed && mousePressed) {
-        context.lineTo(e.pageX - myCanvas.offsetLeft, e.pageY - myCanvas.offsetTop);
-        context.stroke();
+        if (e.target === myCanvas && isOut === false) {
+            context.lineTo(e.pageX - myCanvas.offsetLeft, e.pageY - myCanvas.offsetTop);
+            context.stroke();
+        }
+        else
+            if (e.target !== myCanvas && isOut === false) {
```

Comment:
It is really hard to read it. 

It is 
```
if () {

} else if () {

} else if () {

}
```

or 
```
if () {

} else {
    if () {

    } else if () {

    }
}
```

I know, at the time they are from logic point, but possibility of error is really significant.